.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,45 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  generate-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+      - id: set-matrix
+        run: |
+          find . -type f -name 'go.mod' -printf '%h\n' \
+          | jq -r -R -s 'split("\n")[:-1] | sort | "matrix=\(.)"' >> $GITHUB_OUTPUT
+
+  golangci:
+    needs: generate-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        modules: ${{fromJson(needs.generate-matrix.outputs.matrix)}}
```

Comment:
The downside of this matrix feature is that it clutters up the PR status box:
<img width="683" alt="image" src="https://github.com/cert-manager/cert-manager/assets/978965/505294e9-694f-4a80-852e-f5fb1eec27f8">

I couldn't find a way to only report the overall status of the workflow.